sdks/java/io/cassandra/src/test/java/org/apache/beam/sdk/io/cassandra/CassandraIOTest.java

```diff
@@ -101,7 +101,7 @@
   public static void startCassandra() throws Exception {
     System.setProperty("cassandra.jmx.local.port", JMX_PORT);
     startupTime = System.currentTimeMillis();
-    EmbeddedCassandraServerHelper.startEmbeddedCassandra();
+    EmbeddedCassandraServerHelper.startEmbeddedCassandra(STARTUP_TIMEOUT);
```

Comment:
The default used before was `20_000`. Now this is `45_000`. Does this really prevent the NullPointerException?

Reply:
Should we try to write the test in a defensive way that it fails with a timeout exception instead of a NullPointerException?

Reply:
@mxm The problem is that the failure is in the AfterClass which is launched even if BeforeClass fails: what happens is the cluster startup in BeforeClass fails with a timeout exception consequently Cluster is null and the Cluster.getMetadata() fails in the AfterClass cleaning method. What I can do is avoid calling the cleaning method if Cluster is null.

Reply:
@mxm just pushed defensive code for it that way: 

- if startup time is above 45s, startCassandra BeForeClass fails with an exception and no NPE in AfterClass because it does nothing in that case.
- If startup time is between 20s and 45s then, all the tests are executed and pass.